### PR TITLE
[dbus] enhance external route support

### PR DESCRIPTION
**Pull request by gjc13**
* Add ExternalRoute structure
* Add ExternalRouteTable property
* Enhance tests for the external route.

### CODE REVIEW

File: src/dbus/client/thread_api_dbus.cpp [hunk 1]
```diff
@@ -504,6 +504,11 @@ ClientError ThreadApiDBus::GetRadioTxPower(int8_t &aTxPower)
     return GetProperty(OTBR_DBUS_PROPERTY_RADIO_TX_POWER, aTxPower);
 }
 
+ClientError ThreadApiDBus::GetExternalRouteTable(std::vector<ExternalRoute> &aExternalRouteTable)
```
Comment (bukepo) [inline src/dbus/client/thread_api_dbus.cpp:507]:
```suggestion
ClientError ThreadApiDBus::GetExternalRoutes(std::vector<ExternalRoute> &aExternalRouteTable)
```

File: src/dbus/common/dbus_message_helper_openthread.cpp [hunk 2]
```diff
@@ -146,12 +146,11 @@ otbrError DBusMessageEncode(DBusMessageIter *aIter, const Ip6Prefix &aPrefix)
     VerifyOrExit(dbus_message_iter_open_container(aIter, DBUS_TYPE_STRUCT, nullptr, &sub), error = OTBR_ERROR_DBUS);
     VerifyOrExit(aPrefix.mPrefix.size() <= OTBR_IP6_PREFIX_SIZE, error = OTBR_ERROR_DBUS);
 
-    SuccessOrExit(DBusMessageEncode(&sub, aPrefix.mPrefix));
-    SuccessOrExit(DBusMessageEncode(&sub, aPrefix.mLength));
+    SuccessOrExit(error = DBusMessageEncode(&sub, aPrefix.mPrefix));
+    SuccessOrExit(error = DBusMessageEncode(&sub, aPrefix.mLength));
 
     VerifyOrExit(dbus_message_iter_close_container(aIter, &sub));
```
Comment (wgtdkp) [inline src/dbus/common/dbus_message_helper_openthread.cpp:152]: Set error code.

File: src/dbus/common/dbus_message_helper_openthread.cpp [hunk 3]
```diff
@@ -167,10 +166,48 @@ otbrError DBusMessageExtract(DBusMessageIter *aIter, Ip6Prefix &aPrefix)
...
+
+    VerifyOrExit(dbus_message_iter_open_container(aIter, DBUS_TYPE_STRUCT, nullptr, &sub), err = OTBR_ERROR_DBUS);
+
+    SuccessOrExit(err = DBusMessageEncode(&sub, aRoute.mPrefix));
+    SuccessOrExit(err = DBusMessageEncode(&sub, aRoute.mRloc16));
+    SuccessOrExit(err = DBusMessageEncode(&sub, aRoute.mPreference));
+    SuccessOrExit(err = DBusMessageEncode(&sub, aRoute.mStable));
+    SuccessOrExit(err = DBusMessageEncode(&sub, aRoute.mNextHopIsThisDevice));
+
+    VerifyOrExit(dbus_message_iter_close_container(aIter, &sub));
```
Comment (wgtdkp) [inline src/dbus/common/dbus_message_helper_openthread.cpp:187]: Set error code.

File: tests/dbus/test_dbus_client.cpp [hunk 4]
```diff
@@ -49,6 +53,33 @@ struct DBusConnectionDeleter
 
 using UniqueDBusConnection = std::unique_ptr<DBusConnection, DBusConnectionDeleter>;
 
+static bool operator==(const otbr::DBus::Ip6Prefix &aLhs, const otbr::DBus::Ip6Prefix &aRhs)
+{
+    bool prefixDataEquality = aLhs.mPrefix.size() == aRhs.mPrefix.size() &&
```
Comment (wgtdkp) [inline tests/dbus/test_dbus_client.cpp:58]:
```suggestion
    bool prefixDataEquality = (aLhs.mPrefix.size() == aRhs.mPrefix.size()) &&
```

**Comment by gjc13** [comment]
> is updating openthread submodule necessary?

Accidentally added, already removed.

**Pull request merged**